csharp/src/Apache.Arrow/Column.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using Apache.Arrow.Types;
+
+namespace Apache.Arrow
+{
+    /// <summary>
+    /// A Column data structure that logically represents a column in a dataset
+    /// </summary>
+    public class Column
+    {
+        private readonly Field _field;
+        public Field Field => _field;
+        private readonly ChunkedArray _columnArrays;
+
+        public Column(Field field, IList<Array> arrays)
+        {
+            _columnArrays = new ChunkedArray(arrays);
+            _field = field;
+            if (!Validate())
+            {
+                throw new ArgumentException($"{Field.DataType} must match {_columnArrays.DataType}");
```

Comment:
You may want to rename the `Validate()` method to something like `ValidateTypes()` or similar. If you added a new rule to `Validate` that didn't have to do with the DataTypes, this exception message wouldn't make sense when `Validate()` failed for the new rule.

Reply:
Ah, good point